File: bench/string_builder.ts
// TypeScript peer of bench/string_builder.vader. Idiomatic JS uses an
// array of fragments + final .join(""), mirroring Vader's StringBuilder
// which keeps parts in a string[] and flushes once.

const N = 50_000;
const FRAGMENT = "The quick brown fox jumps over the lazy dog.\n";

const parts: string[] = [];
for (let i = 0; i < N; i++) {
  parts.push(FRAGMENT);
}
const result = parts.join("");
console.log(`string_builder N=${N} len=${result.length}`);

File: bench/string_builder/string_builder.java
// Java peer of bench/string_builder.vader. Uses `java.lang.StringBuilder`,
// the stdlib's pre-allocated-buffer concatenator — closest analogue to
// Vader's `StringBuilder`.

static final int N = 80_000;
static final String FRAGMENT = "The quick brown fox jumps over the lazy dog.\n";

void main() {
    var sb = new StringBuilder();
    for (int i = 0; i < N; i++) {
        sb.append(FRAGMENT);
    }
    var result = sb.toString();
    IO.println("string_builder N=%d len=%d".formatted(N, result.length()));
}
